deploy/acls.crd.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: acls.hoptimator.linkedin.com
+spec:
+  group: hoptimator.linkedin.com
+  names:
+    kind: Acl
+    listKind: AclList
+    plural: acls
+    singular: acl
+    shortNames:
+    - acl
+  preserveUnknownFields: false
+  scope: Namespaced
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          description: Access control rule (colloquially, an Acl)
+          type: object
+          properties:
+            apiVersion:
+              type: string
+            kind:
+              type: string
+            metadata:
+              type: object
+            spec:
+              description: A set of related ACL rules.
+              type: object
+              properties:
+                resource:
+                  description: The resource being controlled.
+                  type: object
+                  properties:
+                    kind:
+                      description: The kind of resource being controlled.
+                      type: string
+                    name:
+                      description: The name of the resource being controlled.
+                      type: string
+                method:
+                  description: The resource access method.
+                  type: string
+                  enum:
+                  - Alter 
+                  - Create
+                  - Delete
+                  - Describe
+                  - Read
+                  - Write
+                  - Post
+                  - Put
+                  - Get
+                  - Head
+                  - Patch
+                  - Trace
+                  - Options
+                  - GetAll
+                  - BatchGet
+                  - BatchCreate
+                  - BatchUpdate
+                  - PartialUpdate
+                  - BatchDelete
+                  - BatchPartialDelete
```

Comment:
Do we have to expose all the options? are we planning to support or need all? Did we take this from somewhere or this is something we think we need to offer?

Reply:
These are all the methods we currently support internally. It's pretty exhaustive but could always be expanded as needed.